syfertext/__init__.py

```diff
@@ -97,3 +113,66 @@ def create(pipeline_name, tags: Set[str] = None, description: str = None):
 
 # Set the default owners of some classes
 # SubPipeline.owner = LOCAL_WORKER
+
+
+def save(pipeline_name: str, pipeline: "Pipeline", destination: Union["local"] = "local") -> None:
+    """Saves the pipeline and it's states to storage
+
+    Args:
+        pipeline_name (str): The name of the pipeline.
+        pipeline (Pipeline): The pipeline object itself
+        destination (Union): The location where to save that object, currently storage on local machine is implemented.
+
+    """
+
+    # Path to the home/SyferText directory
+    data_path = os.path.join(str(Path.home()), "SyferText")
+
+    # Creating a new directory if home/SyferText does not exist
+    if not os.path.exists(data_path):
+        os.mkdir(data_path)
+
+    data_path = os.path.join(data_path, "cache")
+
+    # Creating a new directory if home/SyferText/cache does not exist
+    if not os.path.exists(data_path):
+        os.mkdir(data_path)
+
+    data_path = os.path.join(data_path, pipeline_name)
+
+    # Creating a new directory if home/SyferText/cache/<pipeline_name> does not exist
+    if not os.path.exists(data_path):
+        os.mkdir(data_path)
+
```

Comment:
you can create nested dir using `data_path.mkdir(parents=True, exist_ok=True)`  data_path being pathlib Path object!